### ISSUE TITLE
[Ray Serve]: Testing out on local using Docker container 

### ISSUE THREAD

**Issue by AbhishekBose** [ray-project/ray#28692]
### What happened + What you expected to happen

I am testing out ray serve for the first time on my local environment. Using Docker containers in order to do that

This is the dockerfile which I am using

```
FROM rayproject/ray:latest

COPY requirements.txt /tmp
RUN pip install -r /tmp/requirements.txt

EXPOSE 8000
EXPOSE 8265

WORKDIR /app/

COPY . /app/
ENV PYTHONUNBUFFERED 1

CMD ["bash","entrypoint.sh"]

```

The serving code is given below

```
import ray
from ray import serve
from fastapi import FastAPI
from fastapi import Query
from starlette.responses import JSONResponse
from typing import Dict
import torch
from torchvision import transforms
from onnx_loader import OnnxLoader
import os

app = FastAPI()

@serve.deployment(route_prefix="/")
@serve.ingress(app)
class BarlowTwins:
    def __init__(self) -> None:
        self.model = self.__load_files()
        self.preprocessor = transforms.Compose(
            [transforms.ToTensor()])

    def __load_files(self):
        return OnnxLoader(os.path.join(os.path.dirname(os.path.realpath(__file__)),"barlow_twins_example","assets","barlow.onnx"))

    @app.post("/predict")
    async def predict(self,input_text:str):
        return JSONResponse({
            "prediction":input_text
        })

deployment_graph = BarlowTwins.bind()
```

I am starting the ray server inside the docker container using an entrypoint file.
The commands are as follows

```
ray start --head --node-ip-address=0.0.0.0 --dashboard-host=0.0.0.0
cd barlow_twins_example
serve run server_fast_api:deployment_graph
```

Starting the docker container using `docker compose up`

Output of the docker ps
```

CONTAINER ID   IMAGE                    COMMAND                CREATED          STATUS          PORTS                                            NAMES
02ffd3550d76   ray_container_ray-head   "bash entrypoint.sh"   44 seconds ago   Up 43 seconds   0.0.0.0:8000->8000/tcp, 0.0.0.0:8265->8265/tcp   ray-head
```

The dashboard is visible from my browser and I am to see the barlowTwins deployment

But when I am trying to hit the server at `0.0.0.0:8000`. I am not getting any response. Constantly getting a socket hang up from postman or no response from server when using curl

### Versions / Dependencies

python==3.7.7
ray==2.0.0
fastapi==0.85.0
aiohttp==3.8.1
uvicorn==0.16.0
torch==1.12.1
torchvision==0.13.1
requests==2.28.1
onnxruntime==1.11.1

### Reproduction script

Code snipped has been provided in the description above

### Issue Severity

High: It blocks me from completing my task.

**Comment by cszhu** [comment]
This P2 issue has seen no activity in the past 2 years. It will be closed in 2 weeks as part of ongoing cleanup efforts.

Please comment and remove the `pending-cleanup` label if you believe this issue should remain open.

Thanks for contributing to Ray!
